Cannot compile OpenThread example without the CLI-APP.

#### Description
Cannot compile OpenThread without the cli example feature.

#### Steps to reproduce the issue
On an OpenThread example comment these two lines in the makefile: 
```
USEMODULE += openthread-cli-ftd
USEMODULE += openthread-cli-mtd
```
then do:
`make all`

#### Expected results
Run a Thread instance without the CLI example.

#### Actual results
If doing the procedure on a **clean** example:

```
arm-none-eabi-gcc: error: /home/lab/Documents/TESTRI/RIOT/examples/openthread/bin/samr21-xpro/openthread-ftd.a: No such file or directory
make: *** [/home/lab/Documents/TESTRI/RIOT/examples/openthread/bin/samr21-xpro/openthread.elf] Error 1
```

On the other hand, if commenting the lines in the makefile after a succesfull build with CLI example:

Compiles Ok, but the CLI example remains in the program.

#### Versions

Operating System Environment
-----------------------------
       Operating System: "Ubuntu" "14.04.6 LTS, Trusty Tahr"
                 Kernel: Linux 4.4.0-146-generic i686 i686

Installed compiler toolchains
-----------------------------
             native gcc: gcc (Ubuntu 4.9.4-2ubuntu1~14.04.1) 4.9.4
      arm-none-eabi-gcc: arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 7-2018-q3-update) 7.3.1 20180622 (release) [ARM/embedded-7-branch revision 261907]
                avr-gcc: missing
       mips-mti-elf-gcc: missing
             msp430-gcc: missing
   riscv-none-embed-gcc: missing
                  clang: missing

Installed compiler libs
-----------------------
   arm-none-eabi-newlib: "3.0.0"
    mips-mti-elf-newlib: missing
riscv-none-embed-newlib: missing
               avr-libc: missing (missing)

Installed development tools
---------------------------
                  cmake: cmake version 2.8.12.2
               cppcheck: missing
                doxygen: 1.8.6
                 flake8: missing
                    git: git version 1.9.1
                   make: GNU Make 3.81
                openocd: Open On-Chip Debugger 0.9.0-dev-gbdd8312 (2018-04-05-13:14)
                 python: Python 2.7.6
                python2: Python 2.7.6
                python3: Python 3.4.3
             coccinelle: missing

--- a/examples/openthread/Makefile
+++ b/examples/openthread/Makefile
@@ -13,18 +13,20 @@ RIOTBASE ?= $(CURDIR)/../..
 QUIET ?= 1
 
 USEPKG += openthread
+
+# FTD: A Full Thread Device has router functionality compiled in
+#
+# MTD: A Minimal Thread Device does not have router functionality
+# compiled in. As a result, it is not necessary to configure the
+# routerrole on an MTD. At the same time, an MTD may or may not be sleepy.
+#
+# Use FTD by default
 OPENTHREAD_TYPE ?= ftd
-ifeq ($(OPENTHREAD_TYPE),mtd)
-  # MTD: A Minimal Thread Device does not have router functionality
-  # compiled in. As a result, it is not necessary to configure the
-  # routerrole on an MTD. At the same time, an MTD may or may not be sleepy.
-  USEMODULE += openthread-mtd
-  USEMODULE += openthread-cli-mtd
-else
-  # ftd: A Full Thread Device has router functionality compiled in
-  USEMODULE += openthread-ftd
-  USEMODULE += openthread-cli-ftd
-endif
+
+USEMODULE += openthread-$(OPENTHREAD_TYPE)
+
+# Comment the following line in order to disable the OpenThread CLI
+USEMODULE += openthread-cli
 
 #Define PANID, CHANNEL and UART_BAUDRATE used by default
 OPENTHREAD_PANID ?= 0xbeef

--- a/pkg/openthread/Makefile
+++ b/pkg/openthread/Makefile
@@ -3,15 +3,22 @@ PKG_URL=https://github.com/openthread/openthread.git
 PKG_VERSION=thread-reference-20170716
 PKG_BUILDDIR ?= $(PKGDIRBASE)/$(PKG_NAME)
 
-ifneq (,$(filter openthread-cli-ftd,$(USEMODULE)))
+ifneq (,$(filter openthread-ftd,$(USEMODULE)))
+  TD = ftd
   $(info Compile OpenThread for FTD device)
-  OPENTHREAD_ARGS += --enable-cli-app=ftd
-endif
-ifneq (,$(filter openthread-cli-mtd,$(USEMODULE)))
+else ifneq (,$(filter openthread-mtd,$(USEMODULE)))
   $(info Compile OpenThread for MTD device)
-  OPENTHREAD_ARGS += --enable-cli-app=mtd --enable-joiner
+  TD = mtd
+  JOINER_ARG = --enable-joiner
+else
+  $(error "Please use either USEMODULE=openthread-ftd or USEMODULE=openthread-mtd")
 endif
-OPENTHREAD_ARGS += --enable-application-coap
+
+ifneq (,$(filter openthread-cli,$(USEMODULE)))
+  CLI_ARG = --enable-cli-app=$(TD)
+endif
+
+OPENTHREAD_ARGS += $(CLI_ARG) $(JOINER_ARG) --enable-application-coap
 CONFIG_FILE      = OPENTHREAD_PROJECT_CORE_CONFIG_FILE='\"platform_config.h\"'
 $(info $$OPENTHREAD_ARGS is [${OPENTHREAD_ARGS}])
 
@@ -35,13 +42,10 @@ all: git-download
 	cd $(PKG_BUILDDIR) &&  DESTDIR=$(PKG_BUILDDIR)/output PREFIX=/ make -j4 --no-print-directory install
 
 	cp $(PKG_BUILDDIR)/output/lib/libmbedcrypto.a ${BINDIR}/mbedcrypto.a
-ifneq (,$(filter openthread-cli-ftd,$(USEMODULE)))
-	cp $(PKG_BUILDDIR)/output/lib/libopenthread-ftd.a ${BINDIR}/openthread-ftd.a
-	cp $(PKG_BUILDDIR)/output/lib/libopenthread-cli-ftd.a ${BINDIR}/openthread-cli-ftd.a
-endif
-ifneq (,$(filter openthread-cli-mtd,$(USEMODULE)))
-	cp $(PKG_BUILDDIR)/output/lib/libopenthread-mtd.a ${BINDIR}/openthread-mtd.a
-	cp $(PKG_BUILDDIR)/output/lib/libopenthread-cli-mtd.a ${BINDIR}/openthread-cli-mtd.a
+
+	cp $(PKG_BUILDDIR)/output/lib/libopenthread-$(TD).a ${BINDIR}/openthread-$(TD).a
+ifneq (,$(filter openthread-cli,$(USEMODULE)))
+	cp $(PKG_BUILDDIR)/output/lib/libopenthread-cli-$(TD).a ${BINDIR}/openthread-cli.a
 endif
 
 include $(RIOTBASE)/pkg/pkg.mk

--- a/pkg/openthread/contrib/netdev/openthread_netdev.c
+++ b/pkg/openthread/contrib/netdev/openthread_netdev.c
@@ -73,7 +73,8 @@ static void *_openthread_event_loop(void *arg) {
     netdev_t *dev;
     msg_t msg, reply;
 
-#if defined(MODULE_OPENTHREAD_CLI_FTD) || defined(MODULE_OPENTHREAD_CLI_MTD)
+#if defined(MODULE_OPENTHREAD_CLI)
+    serial_msg_t* serialBuffer;
     otCliUartInit(sInstance);
     /* Init default parameters */
     otPanId panid = OPENTHREAD_PANID;
@@ -91,7 +92,6 @@ static void *_openthread_event_loop(void *arg) {
 #endif
 
     ot_job_t *job;
-    serial_msg_t* serialBuffer;
     while (1) {
         otTaskletsProcess(sInstance);
         if (otTaskletsArePending(sInstance) == false) {
@@ -106,12 +106,14 @@ static void *_openthread_event_loop(void *arg) {
                     dev = msg.content.ptr;
                     dev->driver->isr(dev);
                     break;
+#ifdef MODULE_OPENTHREAD_CLI
                 case OPENTHREAD_SERIAL_MSG_TYPE_EVENT:
                     /* Tell OpenThread about the reception of a CLI command */
                     serialBuffer = (serial_msg_t*)msg.content.ptr;
                     otPlatUartReceived((uint8_t*) serialBuffer->buf,serialBuffer->length);
                     serialBuffer->serial_buffer_status = OPENTHREAD_SERIAL_BUFFER_STATUS_FREE;
                     break;
+#endif
                 case OPENTHREAD_JOB_MSG_TYPE_EVENT:
                     job = msg.content.ptr;
                     reply.content.value = ot_exec_command(sInstance, job->command, job->arg, job->answer);

